### PR TITLE
feat: allow stricter masking

### PR DESCRIPTION
**Pull request by deborahgu**
For some applications, we want to mask all text and images, because non-input text sometimes contains PII. Allows setting this by environment variable.

### CODE REVIEW

**Comment by jsnwesson** [comment]
Oh neat! To clarify, all this does is include an environment variable, but each MFE would have to include their own declaration in `edx-internal`, right? Could you link here the options that Datadog allows for masking?

**Comment by jsnwesson** [comment]
Or maybe include it in frontend-logging's README?

**Review by adamstankiewicz — APPROVED** [review]
I think this approach makes sense to extend `DatadogLoggingService` to allow consumers to override the base `defaultPrivacyLevel`. However, I do wonder if opting to mask _all_ text/copy in a given MFE removes some of the utility/benefits of session replays in terms of debugging and product analytics.

It'd be great if we could mask specific elements (like the username in the header) known to contain PII (e.g., like the username in the MFE's header). See [this Datadog documentation](https://docs.datadoghq.com/real_user_monitoring/session_replay/browser/privacy_options/#override-an-html-element) for example of using `data-dd-privacy` attributes and/or `dd-privacy-*` CSS class names. This is similar to how various PII elements are masked for other tools like Hotjar throughout Open edX currently, e.g., via a `data-hj-suppress` HTML attribute ([examples](https://github.com/search?q=org%3Aopenedx%20data-hj-suppress&type=code)). In fact, the `frontend-app-learner-dashboard` MFE currently masks the username from Hotjar via the `data-hj-suppress` attribute today ([source](https://github.com/openedx/frontend-app-learner-dashboard/blob/9f52c61e4f5b2098a0253a0ba54cf0ff697dde3b/src/containers/LearnerDashboardHeader/ExpandedHeader/AuthenticatedUserDropdown.jsx#L27)).

That said, including tool/vendor specific HTML attributes or class names like this is not really open-source friendly for Open edX since it introduces attributes that are specific to our 2U instance (i.e., not everyone uses Hotjar/Datadog like we do at 2U).

tl;dr; I think supporting the ability for MFEs to override `defaultPrivacyLevel` here makes sense, but it might also be worthwhile to explore how we could extend the Open edX MFE framework to support vendor-specific attributes like `data-hj-suppress` and `data-dd-privacy` in an open-source friendly way so we don't need to mask _everything_ within an MFE just to remove PII from a fairly narrow subset of UI elements, which removes some of the utility of having the session replays from a debugging / product analytics perspective.
